src/include/mallocMC/mallocMC_utils.hpp

```diff
@@ -128,74 +88,64 @@ namespace mallocMC
      *
      * @return current index of the warp
      */
-    MAMC_ACCELERATOR inline auto warpid() -> std::uint32_t
-    {
+    ALPAKA_FN_ACC inline auto warpid()
+    {
+#if defined(__HIP_DEVICE_COMPILE__) && defined(__HIP__)
+        // get wave id
+        // https://github.com/ROCm-Developer-Tools/HIP/blob/f72a669487dd352e45321c4b3038f8fe2365c236/include/hip/hcc_detail/device_functions.h#L974-L1024
+        return __builtin_amdgcn_s_getreg(GETREG_IMMED(3, 0, 4));
+#elif defined(__HIP_DEVICE_COMPILE__) && defined(__HCC__)
```

Comment:
You can remove all `HCC` branches, alpaka is only supporting `HIP-clang` and `HIP-nvcc`
Could you please write `#if defined(__CUDA_ARCH__)` first and HIP as `#elif defined(__HIP_DEVICE_COMPILE__) && defined(__HIP__)`, the result will be that `HIP-nvcc` code performs equal to native CUDA code.

Please apply this change to all utility functions.

Reply:
done